packages/components/src/popover/index.tsx

```diff
@@ -252,6 +253,17 @@ const UnforwardedPopover = (
 		} );
 	}
 
+	let computedVariant = variant;
+	if ( isAlternate !== undefined ) {
+		deprecated( '`isAlternate` prop in wp.components.Popover', {
+			since: '6.2',
+			version: '6.4',
```

Comment:
Technically I think it can't be removed in 6.4, as it's a stable prop and was released in WordPress. 

The same will apply to `anchorRect`, `getAnchorRect` and `anchorRef` on the lines above. 🤔

Reply:
Is that so? Is the `deprecated()` function merely an empty threat? 😂

Just adding a viewpoint on the `isAlternate` thing specifically, but the difference between the default and alternate variants is not something I'd consider a serious visual breakage (see: [tangentially related guideline for style deprecations](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#criteria-for-putting-styles-changes-behind-a-feature-flag)). So personally I think we can either really remove it in 6.4, or if not, we shouldn't be specifying a `version` in the `deprecated()` call at all. (I'm fine with the latter, since it's trivial to support this back compat.)

Reply:
There's some more info in these docs:
- https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/backward-compatibility.md#backward-compatibility
- https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/deprecations.md

(I had no idea that changelog existed in the second document 😱 )

My understanding is:
- Stable APIs that were only released in the plugin can be deprecated and removed/changed after two plugin releases
- Stable APIs that were released in WordPress core are expected to always work, but can be deprecated to encourage devs to migrate to a newer API.
- Unstable / Experimental APIs don't technically have any back compat concerns, but it's usually nice to deprecate them, and give a bit of time for folks to switch away from them.

There are a few things I noticed in popover:
- `__unstableShift` was never released in WordPress core, so didn't need to be deprecated, but it was a nice thing to do. I think the deprecation should have used the plugin's versioning system. I think it can be removed now though, as it has been deprecated for a while.
- All the `anchor` props can't be removed, the best we can do is output a message, make sure they continue to work, and minimize their prominence in the code to keep things tidy.
- Same as above for `range`
- `__unstableForcePosition` was released in WP core. It may have some usage (https://www.wpdirectory.net/search/01GFWMGQ3KH01GQ7CN2DQDN0TC), so I think deprecating for two WP versions is a good move.

I think it might be good to address these things and backport them to WP 6.1 so that there's no confusion. I can put together a PR to do that.

Reply:
Update to the above, it looks like `range` can also be removed, it hasn't ever been a WP api. I made a PR here - https://github.com/WordPress/gutenberg/pull/45195.

Reply:
Will continue the conversation re. deprecations directly in #45195